src/main/java/org/mskcc/cbio/oncokb/web/rest/CronJobController.java

```diff
@@ -182,6 +205,112 @@ public void checkTrialAccounts() {
         mailService.sendTrialAccountExpiresMail(DAYS_TO_CHECK, userDTOS);
     }
 
+    @GetMapping(path = "/check-exposed-tokens")
+    public void exposedTokensInfo() throws IOException, InterruptedException {
+        List<Token> tokens = tokenService
+                                .findAll()
+                                .stream()
+                                .filter(token -> token.getExpiration().isAfter(Instant.now()))
+                                .collect(Collectors.toList());
+        log.info("Searching exposed tokens pipeline begins...");
+        List<ExposedToken> results = new ArrayList<>();
+        for (Token token : tokens){
+            String q = token.getToken().toString();
+
+            int githubCount = 0;
+            try {
+                GitHub github = new GitHubBuilder().withOAuthToken(applicationProperties.getGithubToken()).build();
+                List<GHContent> gitRes = github.searchContent().q(q).list().toList();
+                githubCount = gitRes.size();
+                if (githubCount == 1 && gitRes.get(0).getHtmlUrl().indexOf("https://github.com/knowledgesystems/oncokb-deployment/") == 0){
+                        githubCount = 0;
+                }
+            }
+            catch (Exception e) {
+                e.printStackTrace();
+            }
+
+            int googleCount = 0;
+            try{
+                HttpEntity entity = getGoogleResponse(q).getEntity();
+                Document googleDoc = Jsoup.parse(EntityUtils.toString(entity));
+                Elements resultEle = googleDoc.select("h3");
+                googleCount = resultEle.size();
+                if (googleCount > 0) {
+                    String description = googleDoc.select(".kCrYT > div > div > div > div > div").text();
```

Comment:
I suppose this structure will change in the future. What's our plan if that happens?

Reply:
For google, I have tried its APIs console, but I didn't find a good way to get the same searching result like using google searching. Maybe we could try again in the future to see if I missed some things. But for Baidu, I didn't find it provides any APIs for us to searching. Maybe we can only keep eyes on it or do not index exposed tokens on Baidu. If those structures changed in the future, and there are not better solutions to solve, maybe we can only index tokens on Github.

Reply:
@Leowisd in this case, do you think you can first identify this section exists? If it does not, send an email to the dev team so we can investigate?